third_party/terraform/resources/resource_bigquery_table.go.erb

```diff
@@ -413,9 +584,198 @@ func resourceBigQueryTableDelete(d *schema.ResourceData, meta interface{}) error
+	}
+	if v, ok := raw["csv_options"]; ok {
+		edc.CsvOptions = expandCsvOptions(v)
+	}
+	if v, ok := raw["google_sheets_options"]; ok {
+		edc.GoogleSheetsOptions = expandGoogleSheetsOptions(v)
+	}
+	if v, ok := raw["ignore_unknown_values"]; ok {
+		edc.IgnoreUnknownValues = v.(bool)
+	}
+	if v, ok := raw["max_bad_records"]; ok {
+		edc.MaxBadRecords = int64(v.(int))
+	}
+	if v, ok := raw["schema"]; ok {
+		schema, err := expandSchema(v)
+		if err != nil {
+			return nil, err
+		}
+
+		edc.Schema = schema
+	}
+	if v, ok := raw["source_format"]; ok {
+		edc.SourceFormat = v.(string)
+	}
+
+
+	return edc, nil
+
+}
+
+func flattenExternalDataConfiguration(edc *bigquery.ExternalDataConfiguration) (map[string]interface{}, error) {
+	result := map[string]interface{}{}
+
+	result["autodetect"] = edc.Autodetect
+	result["source_uris"] = edc.SourceUris
+
+	if edc.Compression != "" {
+		result["compression"] = edc.Compression
+	}
+
+	if edc.CsvOptions != nil {
+		result["csv_options"] = flattenCsvOptions(edc.CsvOptions)
+	}
+
+	if edc.GoogleSheetsOptions != nil {
+		result["google_sheets_options"] = flattenGoogleSheetsOptions(edc.GoogleSheetsOptions)
+	}
+
+	if edc.IgnoreUnknownValues == true {
+		result["ignore_unknown_values"] = edc.IgnoreUnknownValues
+	}
+	if edc.MaxBadRecords != 0 {
+		result["max_bad_records"] = edc.MaxBadRecords
+	}
+
+	if edc.Schema != nil {
+		schema, err := flattenSchema(edc.Schema)
+		if err != nil {
+			return result, err
+		}
+		result["schema"] = schema
+	}
+
+	if edc.SourceFormat != "" {
+		result["source_format"] = edc.SourceFormat
+	}
+
+	return result, nil
+}
+
+func expandCsvOptions(configured interface{}) *bigquery.CsvOptions {
+	if len(configured.([]interface{})) == 0 {
+		return nil
+	}
+
+	raw := configured.([]interface{})[0].(map[string]interface{})
+	opts := &bigquery.CsvOptions{}
+
+	if v, ok := raw["allow_jagged_rows"]; ok {
+		opts.AllowJaggedRows = v.(bool)
+	}
+
+	if v, ok := raw["allow_quoted_newlines"]; ok {
+		opts.AllowQuotedNewlines = v.(bool)
+	}
+
+	if v, ok := raw["encoding"]; ok {
+		opts.Encoding = v.(string)
+	}
+
+	if v, ok := raw["field_delimiter"]; ok {
+		opts.FieldDelimiter = v.(string)
+	}
+
+	if v, ok := raw["skip_leading_rows"]; ok {
+		opts.SkipLeadingRows = int64(v.(int))
+	}
+
+	if v, ok := raw["quote"]; ok {
+		quote := v.(string)
+		opts.Quote = &quote
+	}
+
+	return opts
+}
+
+func flattenCsvOptions(opts *bigquery.CsvOptions) []map[string]interface{} {
+	result := map[string]interface{}{}
+
+	if opts.AllowJaggedRows == true {
+		result["allow_jagged_rows"] = opts.AllowJaggedRows
+	}
+
+	if opts.AllowQuotedNewlines == true {
+		result["allow_quoted_newlines"] = opts.AllowQuotedNewlines
+	}
+
+	if opts.Encoding != "" {
+		result["encoding"] = opts.Encoding
+	}
+
+	if opts.FieldDelimiter != "" {
+		result["field_delimiter"] = opts.FieldDelimiter
+	}
+
+	if opts.SkipLeadingRows != 0 {
+		result["skip_leading_rows"] = opts.SkipLeadingRows
+	}
+
+	if *opts.Quote != "" {
```

Comment:
> If your data does not contain quoted sections, set the property value to an empty string. 

Based on that part of the docs, I think we need to allow the empty string. We should also set `ForceSendFields` to include this field, so that we send the empty string when it's set. The client library we're using strips zero values by default when sending requests.

Reply:
Ok, I'll try that.